### ISSUE TITLE
openCL delegate generates '0' and 'inf' values with reduce_sum

### ISSUE THREAD

**Issue by Bahar-BM** [tensorflow/tensorflow#56732]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

tf 2.8, tf 2.9

### Custom Code

No

### OS Platform and Distribution

Android

### Mobile device

tested on Snapdragon 888, 865

### Current Behaviour?

```shell
My models with 'tf.math.reduce_sum' ('Sum' in the tflite version) nodes generate wrong results with the openCL delegate. My experiments show that the output of a 'Sum' node in a tflite model contains lots of 'zeros' and 'inf' values when we use openCL delegate. This issue does not happen with other delegates like XNNPACK.
```

### Standalone code to reproduce the issue

```shell
Here is a very simple model with a reduce_sum node in the structure:

x0 =  Input(shape=(23, 512))
x1 = tf.math.reduce_sum(x0, axis=-1)

model = Model([x0], [x1], name='test')
model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
model.save('reduce_sum.h5')

The tflite version of this model generates zeros and inf values with the openCL delegate.
```

### Relevant log output

_No response_</details>

**Comment by mohantym** [comment]
Hi @Bahar-BM !
Sorry for the late response.  Could you share complete code including TFlite conversion and inference to replicate the issue.
Thank you!

**Comment by you74674** [comment]
I have similar issue with reduce_mean using OpenCL delegate.
for a (1 x 2 x 2 x 1) input, reduce across axis=2 gives wrong result. axis=1 or axis=(1,2) is fine.
also with keepDims=True the result is correct, so I think something might be wrong when keepDims=False. 
(I didn't check for other input size, though)

**Comment by Bahar-BM** [comment]
Hi @mohantym, we have implemented a small tool for you to reproduce the mentioned issue with the reduce_sum node. Here is the link to the repository:
https://github.com/Bahar-BM/tflite-test
In this repository we have also included scripts for building and converting the sample model.

**Comment by mohantym** [comment]
Hi @sachinprasadhs !
Could you please look at this issue.
Thank you!

**Comment by roserg** [comment]
Hi,
Can you check that your models work correct after this commit:
https://github.com/tensorflow/tensorflow/commit/56a417633249bef47f94b2754687731e228758d6

**Comment by Bahar-BM** [comment]
Hi @roserg,
Thank you for your response. Yes! your fix has resolved the problem with the "reduce_sum" node. Thank you!
However, I noticed that there are other nodes like "tf.stack" with the same issue. Should I close this issue and open a new one for the other node?

**Comment by sachinprasadhs** [comment]
@Bahar-BM , If the issue is fixed with the latest nightly version, could you please close this request.
You can open the new request for the above commented issue. Thanks!

**Comment by Bahar-BM** [comment]
@sachinprasadhs Yes, commit [56a4176](https://github.com/tensorflow/tensorflow/commit/56a417633249bef47f94b2754687731e228758d6) has resolved the issue with the "reduce_sum" node. So, I'll close this issue and will open a new one for the other similar nodes.
